src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java

```diff
@@ -1255,6 +1285,13 @@ public JDTTreeBuilder(Factory factory) {
 		}
 		CtTypeReference<?> currentReference = typeReference;
 
+		setPositionReference(typeReference, type);
+		int sourceStartDeclaration = type.sourceStart();
+		int sourceStartSource = type.sourceStart();
+		int sourceEnd = type.sourceEnd();
+		CompilationUnit cu = factory.CompilationUnit().create(new String(context.compilationunitdeclaration.getFileName()));
+		typeReference.setPosition(factory.Core().createSourcePosition(cu, sourceStartDeclaration, sourceStartSource, sourceEnd, context.compilationunitdeclaration.compilationResult.lineSeparatorPositions));

```

Comment:
You can't put the reference in the context to do that in the `JDTTreeBuilder#enter` method?